packages/node-opentelemetry/src/sentry/node.ts

```diff
@@ -0,0 +1,114 @@
+import api, { Span, SpanKind } from '@opentelemetry/api';
+import { ExceptionEventName } from '@opentelemetry/sdk-trace-base/build/src/enums';
+import {
+  SEMATTRS_EXCEPTION_MESSAGE,
+  SEMATTRS_EXCEPTION_STACKTRACE,
+  SEMATTRS_EXCEPTION_TYPE,
+} from '@opentelemetry/semantic-conventions';
+import { Event, EventHint, Exception } from '@sentry/types';
+
+import hdx from '../debug';
+import { jsonToString } from '../utils';
+
+const tracer = api.trace.getTracer('@hyperdx/node-opentelemetry');
+
+// CUSTOM SEMANTIC CONVENTIONS
+const SEMATTRS_EXCEPTION_MODULES = 'exception.modules';
+const SEMATTRS_EXCEPTION_TAGS = 'exception.tags';
+
+// https://github.com/open-telemetry/opentelemetry-js/blob/ca027b5eed282b4e81e098ca885db9ce27fdd562/packages/opentelemetry-sdk-trace-base/src/Span.ts#L299
+const recordException = (span: Span, exception: Exception) => {
+  span.addEvent(ExceptionEventName, {
+    [SEMATTRS_EXCEPTION_MESSAGE]: exception.value,
+    [SEMATTRS_EXCEPTION_STACKTRACE]: jsonToString(exception.stacktrace),
+    [SEMATTRS_EXCEPTION_TYPE]: exception.type,
+  });
+};
+
+const isSentryEventAnException = (event: Event) =>
+  event.exception?.values?.length > 0;
+
+// TODO: enrich span with more info
+const buildSingleSpanName = (event: Event) =>
+  [event.exception?.values[0].type, event.transaction].join(' ');
+
+const startOtelSpanFromSentryEvent = (event: Event, hint: EventHint) => {
+  // FIXME: can't attach to the active span
+  // since Sentry would overwrite the active span
+  // let span = api.trace.getActiveSpan();
```

Comment:
we can't attach exception events to active span since the downstream (likely Sentry client) would reset it. For some reasons, when I tested it, the 'addEvent' api doesn't work properly